mempool/clist_mempool.go

```diff
@@ -267,7 +267,7 @@ func (mem *CListMempool) CheckTx(tx types.Tx, sender p2p.ID) (*abcicli.ReqRes, e
 				memTx := elem.Value.(*mempoolTx)
 				if found := memTx.addSender(sender); found {
 					// It should not be possible to receive twice a tx from the same sender.
-					mem.logger.Error("tx already received from peer", "tx", tx.Hash(), "sender", sender)
+					mem.logger.Error("tx already received from peer", "tx", fmt.Sprintf("%X", tx.Hash()), "sender", sender)
```

Comment:
Here `log.NewLazySprintf` is better than `fmt.Sprintf`. This also applies to many of the other changes below. Also, `tx.Hash()` is just `tx.Key()[:]`. `Key` is called four lines above.